internal/cmd/controller/reconciler/cluster_controller.go

```diff
@@ -100,6 +101,7 @@ func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			}),
 		).
 		WithEventFilter(sharding.FilterByShardID(r.ShardID)).
+		WithOptions(controller.Options{MaxConcurrentReconciles: 50}).
```

Comment:
Now I think that at least this one should also be made configurable.

Reply:
Yes, that's a good follow-up story.